### ISSUE TITLE
Get result for router.go()

### ISSUE THREAD

**Issue by yunyu** [vuejs/vue-router#1705]
### What problem does this feature solve?
I have a /login page that should redirect the user to the previous page upon logging in. Using router.go(-1) works in most situations, except when you directly visit the /login page without going through the home page. 

### What does the proposed API look like?
There should be a way to get the result of router.go() and redirect to a default page if it fails.

    router.go(-1, result => if (!result.success) router.replace('/'))

**Comment by posva** [comment]
We have the navigation guards for that purporse: https://router.vuejs.org/en/advanced/navigation-guards.html

**Comment by yunyu** [comment]
@posva I can't seem to figure out how to get this to work. I've added a beforeResolve to the global router hook, and it seems to redirect even when I don't call next() when the from path is /login. I've also tried adding a component level beforeRouteLeave, but I get the exact same results. For example, even this code (which should cause the router to never change) goes back to the previous page (in my case, the new tab page): 

    beforeRouteLeave(to, from, next) {
        console.log(to)
        next(false)
    }

The above cancels the navigation when I visit a page in my SPA before /login is hit, but has zero effect when /login is the first page hit (the browser still navigates back to new tab).
